geopyspark/geotrellis/rdd.py

```diff
@@ -478,3 +478,39 @@ def reclassify(self, value_map, data_type, boundary_strategy=LESSTHANOREQUALTO):
 
         srdd = _reclassify(self.srdd, value_map, data_type, boundary_strategy)
         return TiledRasterRDD(self.geopysc, self.rdd_type, srdd)
+
+    def _process_operation(self, value, operation):
+        if isinstance(value, int) or isinstance(value, float):
+            srdd = operation(value)
+        elif isinstance(value, TiledRasterRDD):
+            if self.rdd_type != value.rdd_type:
+                raise ValueError("Both TiledRasterRDDs need to have the same rdd_type")
+            srdd = operation(value.srdd)
```

Comment:
Actually since we have the metadata we should really check that the CRS and layouts themselves are the same. The only tricky part is that CRS checking needs to be clever since they're are multiple ways to specify equivalent CRS.